sources/packages/api/src/services/institution/institution.service.ts

```diff
@@ -655,4 +659,37 @@ export class InstitutionService extends RecordDataModelService<Institution> {
     institution.notes.push(note);
     await this.repo.save(institution);
   }
+
+  /**
+   * Service to get the admin roles .
+   * @returns Admin roles.
+   */
+  async getAdminRoles(): Promise<InstitutionUserTypeAndRole[]> {
+    return this.institutionUserTypeAndRoleRepo
+      .createQueryBuilder("userRole")
+      .select(["userRole.type", "userRole.role"])
+      .where("userRole.type = 'admin'")
+      .getMany();
+  }
+
+  /**
+   * Checks if a legal signing authority is assigned to an Institution.
+   * @param institutionId
+   * @returns Legal signing authority object.
+   */
+  async checkLegalSigningAuthority(
+    institutionId: number,
+  ): Promise<InstitutionUserAuth> {
+    const query = this.institutionUserAuthRepo
+      .createQueryBuilder("userAuth")
+      .select("userAuth.id")
+      .innerJoin("userAuth.institutionUser", "institutionUser")
+      .innerJoin("userAuth.authType", "role")
+      .innerJoin("institutionUser.institution", "institution")
+      .where("institution.id = :institutionId", { institutionId })
+      .andWhere("role.role = :legalSigningAuthority", {
+        legalSigningAuthority: InstitutionUserRoles.legalSigningAuthority,
+      });
+    return query.getOne();
```

Comment:
do we have a check already in place, that only one user in the institution will have the the legal-signing-authority role?

Reply:
Yeah the requirement is only one legal signing authority can be assigned for an institution and controller validates this on every create and update operation.